Feature/smhe 1698 dlmsconfigservice read alarm register command executor

natasja-n opened the pull request:
Use dlms config service in the ReadAlarmRegisterCommandExecutor and add unit tests.

https://jenkins.fdp.osgp.cloud/view/SMHE-cucumber/job/gxf-smhe-cucumber-build/94/

natasja-n committed:
w.i.p. add multipart config to feign api client

Signed-off-by: Natasja Nortier <natasja.nortier@alliander.com>

natasja-n committed:
Merge branch 'development' of github.com:OSGP/open-smart-grid-platform into development

natasja-n committed:
Merge branch 'development' of github.com:OSGP/open-smart-grid-platform into development

natasja-n committed:
Merge branch 'development' of github.com:OSGP/open-smart-grid-platform into development

natasja-n committed:
Merge branch 'development' of github.com:OSGP/open-smart-grid-platform into development

natasja-n committed:
Merge branch 'development' of github.com:OSGP/open-smart-grid-platform into development

natasja-n committed:
SMHE-1564 uitvragen van een IP-adres uit de LTe-m Radius & stub, obv zsmart interface - smhe aanpassing
add cucumber test and handle bundle when failed

Signed-off-by: Natasja <natasja.nortier@alliander.com>

natasja-n committed:
Merge branch 'development' of github.com:OSGP/open-smart-grid-platform into development

natasja-n committed:
SMHE-1564 Radius add event
fix cucumber test

Signed-off-by: Natasja Nortier <natasja.nortier@alliander.com>

natasja-n committed:
Merge branch 'development' of github.com:OSGP/open-smart-grid-platform into development

natasja-n committed:
Merge branch 'development' of github.com:OSGP/open-smart-grid-platform into development

natasja-n committed:
Merge branch 'development' of github.com:OSGP/open-smart-grid-platform into development

natasja-n committed:
Merge branch 'development' of github.com:OSGP/open-smart-grid-platform into development

natasja-n committed:
SMHE-1689 Use dlms config service for ReadAlarmRegisterCommandExecutor
use dlms config service and add unit tests

Signed-off-by: Natasja <natasja.nortier@alliander.com>

natasja-n committed:
Merge branch 'development' of github.com:OSGP/open-smart-grid-platform into development

natasja-n committed:
Merge branch 'development' into feature/SMHE-1698_dlmsconfigservice_ReadAlarmRegisterCommandExecutor

natasja-n committed:
SMHE-1689 Use dlms config service for ReadAlarmRegisterCommandExecutor
use dlms config service and add unit tests

Signed-off-by: Natasja <natasja.nortier@alliander.com>

natasja-n committed:
SMHE-1689 Use dlms config service for ReadAlarmRegisterCommandExecutor
use dlms config service and add unit tests

Signed-off-by: Natasja <natasja.nortier@alliander.com>

natasja-n committed:
SMHE-1698 Use dlms config service for ReadAlarmRegisterCommandExecutor
extend cucumber tests ReadAlarmRegister with different protocols and versions

Signed-off-by: Natasja Nortier <natasja.nortier@alliander.com>

kroesctrl commented on integration-tests/cucumber-tests-platform-smartmetering/src/test/resources/features/osgp-adapter-ws-smartmetering/bundle/BundledReadAlarmRegister.feature:
```diff
@@ -8,13 +8,24 @@ Feature: SmartMetering Bundle - ReadAlarmRegister
+      | SelectiveAccessSupported | true              |
+      | Protocol                 | <protocol>        |
+      | ProtocolVersion          | <version>         |
+      | Port                     |              1028 |
```
Doordat je poort 1028 gebruikt, test hij alleen maar op een 5.1 simulator, deze poort is afahankelijk van het protocol. Als je hem niet specificeert dan pakt de test automatisch de juiste. 
De port moet hier iig niet fixed op 1028 staan

natasja-n replied:
ok, port is aangepast.

stefanermens commented on osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/alarm/ReadAlarmRegisterCommandExecutor.java:
```diff
@@ -37,13 +38,16 @@ public class ReadAlarmRegisterCommandExecutor
 
-  final DlmsObjectConfigService dlmsObjectConfigService;
+  final ObjectConfigServiceHelper objectConfigServiceHelper;
+  static final int ALARM_REGISTER_1_ATTRIBUTE_ID = 2;
```
Alarm register 1 and 2 are both of type DLMS class Register. So the attribute id is the same for both. To avoid the magic number, you could use RegisterAttribute.VALUE.attributeId().

natasja-n replied:
The attribute id value is updated to RegisterAttribute.VALUE.attributeId().

stefanermens commented on osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/alarm/ReadAlarmRegisterCommandExecutorTest.java:
```diff
@@ -0,0 +1,304 @@
+
+  private static final String OBIS_CODE_ALARM_REGISTER_1 = "0.0.97.98.0.255";
+  private static final String OBIS_CODE_ALARM_REGISTER_2 = "0.0.97.98.1.255";
+  private static final int CLASS_ID_READ_ALARM_REGISTER = 1;
```
The class id of the register is 3. You could take that number from InterfaceClass.REGISTER.id()

natasja-n replied:
The class id value is updated to InterfaceClass.REGISTER.id().

stefanermens commented on osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/alarm/ReadAlarmRegisterCommandExecutorTest.java:
```diff
@@ -0,0 +1,304 @@
+
+    final ObisCode newObisCode = new ObisCode(obisCode);
+
+    final AttributeAddress attributeAddress = mock(AttributeAddress.class);
```
Is it necessary to create a mock? Suggestion:
`... = new AttributeAddress(InterfaceClass.REGISTER.id(), new ObisCode(obisCode), RegisterAttribute.VALUE.attributeId());`

natasja-n replied:
The mock is replaced by new AttributeAddress(InterfaceClass.REGISTER.id(), new ObisCode(obisCode), RegisterAttribute.VALUE.attributeId());

natasja-n committed:
SMHE-1698 Use dlms config service for ReadAlarmRegisterCommandExecutor
review: remove port from cucumber test, update attributeid, use attributeAddress instead of mock in unit test

Signed-off-by: Natasja <natasja.nortier@alliander.com>

natasja-n committed:
Merge branch 'development' of github.com:OSGP/open-smart-grid-platform into development

natasja-n committed:
Merge branch 'development' of github.com:OSGP/open-smart-grid-platform into development

natasja-n committed:
Merge branch 'development' into feature/SMHE-1698_dlmsconfigservice_ReadAlarmRegisterCommandExecutor

natasja-n committed:
SMHE-1698 Use dlms config service for ReadAlarmRegisterCommandExecutor
update cucumber test

Signed-off-by: Natasja <natasja.nortier@alliander.com>

kroesctrl commented on integration-tests/cucumber-tests-platform-smartmetering/src/test/resources/features/osgp-adapter-ws-smartmetering/bundle/BundledReadAlarmRegister.feature:
```diff
@@ -7,14 +7,25 @@ Feature: SmartMetering Bundle - ReadAlarmRegister
     Then the bundle response should contain a read alarm register response
+    Examples:
+      | protocol | version | port | lls1active | hls5active |
+      | DSMR     | 2.2     | 1026 | true       | false      |
```
Volgens mij is de nieuwe conventie dat we de poort niet meer specificeren, deze wordt automatcisch bepaald op basis van het protocol

natasja-n replied:
correct, port is er uit gehaald.

natasja-n committed:
SMHE-1698 Use dlms config service for ReadAlarmRegisterCommandExecutor
remove port from test

Signed-off-by: Natasja <natasja.nortier@alliander.com>

natasja-n committed:
Merge branch 'development' of github.com:OSGP/open-smart-grid-platform into development

natasja-n committed:
Merge branch 'development' of github.com:OSGP/open-smart-grid-platform into development

natasja-n committed:
Merge branch 'development' into feature/SMHE-1698_dlmsconfigservice_ReadAlarmRegisterCommandExecutor

natasja-n committed:
SMHE-1698 Use dlms config service for ReadAlarmRegisterCommandExecutor
add smr 5.2 and 5.5, update ALARM_REGISTER_2 and 3 in AlarmTypeRegisterLookup

Signed-off-by: Natasja <natasja.nortier@alliander.com>

natasja-n committed:
SMHE-1698 Use dlms config service for ReadAlarmRegisterCommandExecutor
add smr 5.2 and 5.5 in cucumber test, update alarmobject.register2.value and alarmobject.register3.value

Signed-off-by: Natasja <natasja.nortier@alliander.com>

natasja-n commented:
5.2 en 5.5 worden nu ook afgetest.
